source/config/profileUpgrader.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: UTF-8 -*-
+#A part of NonVisual Desktop Access (NVDA)
+#Copyright (C) 2016 NV Access Limited
+#This file is covered by the GNU General Public License.
+#See the file COPYING for more details.
+
+from logHandler import log
+from configSpec import latestSchemaVersion, confspec
+from configobj import flatten_errors
+from copy import deepcopy
+import profileUpgradeSteps
+
+schemaVersionKey = "schemaVersion"
+
+def upgrade(profile, validator):
+	# when profile is none or empty we can still validate. It should at least have a version set.
+	log.info("Checking schema version of config:\n%s" % profile)

```

Comment:
Sorry. My bad; didn't read it properly. I was confused because I saw "Checking schema version of config" and assumed it was just going to be a version number or something, not the config itself. If I'd looked more closely, I would have seen the profile variable afterwards. :)

I am a bit dubious about logging the entire config all the time. I totally understand why we'd log it before an upgrade, but unless I'm missing something, this will get logged on every load. It's true that having the config might be useful for debugging in some cases, but I feel it'd be appropriate for us to provide instructions to the user in that case; e.g. setting to level debug.